src/main/java/io/neonbee/internal/codec/DataExceptionMessageCodec.java

```diff
@@ -0,0 +1,79 @@
+package io.neonbee.internal.codec;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import io.neonbee.data.DataException;
+import io.netty.util.CharsetUtil;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.eventbus.MessageCodec;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+public class DataExceptionMessageCodec implements MessageCodec<DataException, DataException> {
+
+    @Override
+    public void encodeToWire(Buffer buffer, DataException exception) {
+        buffer.appendInt(exception.failureCode());
+
+        String msgEncoded = exception.getMessage();
+        if (msgEncoded != null) {
+            byte[] msgBytes = msgEncoded.getBytes(CharsetUtil.UTF_8);
+            buffer.appendInt(msgBytes.length);
+            buffer.appendBytes(msgBytes);
+        } else {
+            buffer.appendInt(-1);
+        }
+
+        if (exception.failureDetail().isEmpty()) {
+            buffer.appendInt(-1);
+        } else {
```

Comment:
Is this required? An empty JsonObject should be encoded as {}, so we do not need special logic for it :)